lib/trento/application/usecases/hosts/hosts.ex

```diff
@@ -36,4 +47,41 @@ defmodule Trento.Hosts do
         subscription_count
     end
   end
+
+  @spec deregister_host(Ecto.UUID.t(), DateService) :: :ok | {:error, any}
+  def deregister_host(host_id, date_service \\ DateService) do
+    case Repo.get_by(HostReadModel, id: host_id) do
+      nil ->
+        {:error, :host_not_found}
+
+      _ ->
+        maybe_dispatch_host_deregistration_request(host_id, date_service)
+    end
+  end
+
+  defp commanded,
+    do: Application.fetch_env!(:trento, Trento.Commanded)[:adapter]
+
+  defp maybe_dispatch_host_deregistration_request(host_id, date_service) do
+    now = date_service.utc_now()
+    total_deregistration_debounce = @heartbeat_interval + @deregistration_debounce
```

Comment:
Could we maybe move this to a constant above? No need to compute it every time.
Wdyt?